frappe/core/doctype/user/user.py

```diff
@@ -182,6 +182,13 @@ def validate(self):
 
 		if (self.name not in ["Administrator", "Guest"]) and (not self.get_social_login_userid("frappe")):
 			self.set_social_login_userid("frappe", frappe.generate_hash(length=39))
+		self.disable_email_fields_if_user_disabled()
```

Comment:
We have a function that's called for similar purposes already: https://github.com/frappe/frappe/blob/f7d04dd35c6f5d04ef8853867a82219c60db82e4/frappe/core/doctype/user/user.py#L168

So you can add a check in there